plugins/device/store-utils.go

```diff
@@ -13,16 +14,48 @@ func GetDevices(peerID string) ([]deviceapi.Info, error) {
 	if err != nil {
 		return nil, err
 	}
-	if len(peerInfo.MetaData["devices"]) > 0 {
+	if len(peerInfo.Metadata["devices"]) > 0 {
 		var deviceInfo []deviceapi.Info
-		if err := json.Unmarshal([]byte(peerInfo.MetaData["devices"]), &deviceInfo); err != nil {
+		if err := json.Unmarshal([]byte(peerInfo.Metadata["devices"]), &deviceInfo); err != nil {
 			return nil, err
 		}
 		return deviceInfo, nil
 	}
 	return nil, nil
 }
 
+//CheckIfDeviceExist returns error if all devices already exist or returns list of devices to be added
```

Comment:
`CheckIfDeviceExist()` can be simplified to just return a bool.
The Unmarshalling can be moved out to the caller.